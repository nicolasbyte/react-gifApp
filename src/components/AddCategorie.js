import React, { useState } from 'react';
import PropTypes from 'prop-types';


export const AddCategorie = ( { setCategories } ) => {

    const [categorie, setCategorie] = useState('');

    const addHandle = ( e ) => {
        
        setCategorie( e.target.value )
    }

    const addSumbit = ( e ) => {
        e.preventDefault();

        if ( categorie.trim().length > 2) {
            
            setCategories( cat => [ categorie , ...cat ] );
            setCategorie( '' );
        }


    }
    return (
        
            <form onSubmit = { addSumbit } >
                <p> { categorie } </p>
                <input
                        type     = "text"
                        value    = { categorie }
                        onChange = { addHandle }
                    />
            </form>

    );

};


AddCategorie.propTypes = {

    setCategories : PropTypes.func.isRequired 
}
