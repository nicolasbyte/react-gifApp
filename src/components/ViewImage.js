import React from 'react';
import PropTypes from 'prop-types';


export const ViewImage = ( { title , url } ) => {
    return (
        <div className = "card animate__animated animate__fadeIn">
                <img src = { url } alt = { title } />
                <p> { title } </p>           
        </div>
    );
};


ViewImage.propTypes = {
    title : PropTypes.string.isRequired,
    url   : PropTypes.string.isRequired
};
