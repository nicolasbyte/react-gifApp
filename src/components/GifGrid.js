import React from 'react';
import PropTypes from 'prop-types';

import { useGifFetch } from '../hooks/useGifFetch';
import { ViewImage } from './ViewImage';

export const GifGrid =  ( { category } ) => {

    
    const { loading , data : images } = useGifFetch( category );



    
    

    return (
        <>
        <h3> { category } </h3>
        { loading && <p>Cargando...</p> }
        <div className = "card-grid animate__animated animate__fadeIn">
            
                {
                    images.map(  img  => (
                        <ViewImage 
                                    key = { img.id }
                                    { ...img } />
                    ) )
                }
            
        </div>
        </>
    );
}


GifGrid.propTypes = {
    category : PropTypes.string.isRequired
}