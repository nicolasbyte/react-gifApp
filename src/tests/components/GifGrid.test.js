import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useGifFetch } from "../../hooks/useGifFetch";
jest.mock('../../hooks/useGifFetch');

describe(' pruebas en el componente de GifGrid', () => {
    const category = 'One Punch';
    
    test('hacer snapshot con contra el GifGrid', () => {
        
        useGifFetch.mockReturnValue({
            data    : [],
            loading : true
        });
        const wrapper = shallow( <GifGrid  category = { category } />)
        expect( wrapper ).toMatchSnapshot();
    });
    
    test('debe de traer informaciÃ³n', () => {
        
        const data = [
            {
                id    : 'ABC',
                url   : 'htttp://localhost/cualquier/cosa',
                title : 'Cualquier Cosa'
            }
        ];
        
        useGifFetch.mockReturnValue({
            data,
            loading : false
        });
        
        const wrapper = shallow( <GifGrid  category = { category } />)
        expect( wrapper ).toMatchSnapshot();

        expect( wrapper.find('p').exists() ).toBe( false );
        expect( wrapper.find('ViewImage').length ).toBe( data.length );

    });
    
    
});
