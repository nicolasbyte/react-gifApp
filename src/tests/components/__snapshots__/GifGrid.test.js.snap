// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` pruebas en el componente de GifGrid debe de traer informaci√≥n 1`] = `
<Fragment>
  <h3>
     
    One Punch
     
  </h3>
  <div
    className="card-grid animate__animated animate__fadeIn"
  >
    <ViewImage
      id="ABC"
      key="ABC"
      title="Cualquier Cosa"
      url="htttp://localhost/cualquier/cosa"
    />
  </div>
</Fragment>
`;

exports[` pruebas en el componente de GifGrid hacer snapshot con contra el GifGrid 1`] = `
<Fragment>
  <h3>
     
    One Punch
     
  </h3>
  <p>
    Cargando...
  </p>
  <div
    className="card-grid animate__animated animate__fadeIn"
  />
</Fragment>
`;
